### PR TITLE
torch.nn.Unfold accepts 0-dim for batch size

### PR DESCRIPTION
**Pull request by v0dro**
In partial completion of https://github.com/pytorch/pytorch/issues/12013

Allows specifying a tensor with 0-dim batch size for `torch.nn.Unfold()`.

### CODE REVIEW

**Comment by v0dro** [comment]
@rgommers could you please review this one?

File: aten/src/ATen/native/im2col_shape_check.h [hunk 1]
```diff
@@ -37,9 +37,11 @@ static inline void col2im_shape_check(
       dilation_width);
 
   int64_t ndim = input.ndimension();
+  // allow dim=0 only the batch dimension.
   TORCH_CHECK(
-      input.numel() != 0 && (ndim == 2 || ndim == 3),
-      "Expected non-empty 2D or 3D input tensor, but got input of sizes",
+      (ndim == 2 && input.size(1) != 0) ||
```
Comment (ngimel) [inline aten/src/ATen/native/im2col_shape_check.h:42]: same as in replication pad, if ndim==2 you need size(1) and size(0) both to be non-zero, similar for other checks

Comment (v0dro) [reply aten/src/ATen/native/im2col_shape_check.h:42]: Ah I see.. thanks for that I'll send a patch soon.

**Comment by v0dro** [comment]
@ngimel request review.

**Comment by v0dro** [comment]
Any update on this?

**Review by ngimel — APPROVED** [review]
Thanks!
